browser/admin/src/integrator/AdminIntegratorSettings.ts

```diff
@@ -417,6 +438,388 @@ class SettingIframe {
+						}
+						materialIcon.innerHTML = `
+							<svg fill="currentColor" class="material-design-icon__svg" width="24" height="24" viewBox="0 0 24 24">
+							${
+								currentChecked
+									? `<path d="M10,17L5,12L6.41,10.58L10,14.17L17.59,6.58L19,8M19,3H5C3.89,3 3,3.89 3,5V19A2,2 0 0,0 5,21H19A2,2 0 0,0 21,19V5C21,3.89 20.1,3 19,3Z">
+									<!---->
+								</path>`
+									: `<path d="M19,3H5C3.89,3 3,3.89 3,5V19A2,2 0 0,0 5,21H19A2,2 0 0,0 21,19V5C21,3.89 20.1,3 19,3M19,5V19H5V5H19Z">
+									<!---->
+								</path>`
+							}
+							</svg>`;
+
+						data[key] = currentChecked;
+					});
+
+					container.appendChild(checkboxWrapper);
+				}
+			}
+		}
+		return container;
+	}
+
+	private collectBrowserSettingsFromUI(
+		browserSettingSection: HTMLElement,
+	): void {
+		const inputs = browserSettingSection.querySelectorAll<HTMLInputElement>(
+			'input.checkbox-radio-switch-input',
+		);
+
+		inputs.forEach((input) => {
+			// Expected ID: section-setting-input (e.g., "writer-ShowSidebar-input")
+			const parts = input.id.split('-');
+			if (parts.length !== 3 || parts[2] !== 'input') return;
+
+			const [sectionRaw, settingKey] = parts;
+			const value = input.checked;
+
+			if (sectionRaw === 'common') {
+				this.browserSettingOptions[settingKey] = value;
+			} else {
+				(this.browserSettingOptions[sectionRaw] as Record<string, boolean>)[
+					settingKey
+				] = value;
+			}
+		});
+	}
+
+	private customRenderers: Record<
+		string,
+		(key: string, value: any, uniqueId: string) => HTMLElement
+	> = {
+		compactToggle: this.renderCompactModeToggle.bind(this),
+	};
+
+	private renderCompactModeToggle(
+		key: string,
+		setting: any,
+		uniqueId: string,
+	): HTMLElement {
+		const container = document.createElement('div');
+		container.className = 'custom-compact-toggle';
+
+		const inputCheckbox = document.createElement('input');
+		inputCheckbox.type = 'checkbox';
+		inputCheckbox.className = 'checkbox-radio-switch-input';
+		inputCheckbox.id = uniqueId + '-input';
+		inputCheckbox.checked = setting.value;
+		inputCheckbox.style.display = 'none'; // hidden input for logic
+		container.appendChild(inputCheckbox);
+
+		const options = document.createElement('div');
+		options.className = 'toggle-options'; // You can style with flex and gap
+
+		// Create option for Notebookbar
+		const notebookOption = document.createElement('div');
+		notebookOption.className = 'toggle-option';
+		const notebookImage = document.createElement('img');
+		notebookImage.src = 'images/Notebookbar.svg';
+		notebookImage.alt = 'Notebookbar';
+		notebookImage.className = `toggle-image ${!setting.value ? 'selected' : ''}`;
+		const notebookLabel = document.createElement('div');
+		notebookLabel.textContent = _('Notebookbar view');
+		notebookLabel.className = 'toggle-image-label';
+		notebookOption.appendChild(notebookImage);
+		notebookOption.appendChild(notebookLabel);
+
+		// Create option for Compact mode
```

Comment:
can we share more code? these two blocks are identical, only "Notebookbar" can be changed with "Compact" :)